### PR TITLE
New Themed Wallet List Modal/Picker

### PR DESCRIPTION
**Pull request by amanalili**
#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android

Screen Shot 2021-04-05 at 6 24 41 PM

Screen Shot 2021-04-05 at 6 24 56 PM

Screen Shot 2021-04-05 at 6 25 49 PM

Screen Shot 2021-04-05 at 6 26 03 PM

### CODE REVIEW

File: src/components/themed/WalletList.js [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
-import { WalletListRow } from './WalletListRow.js'
-
-type WalletListItem = { id: string, fullCurrencyCode?: string, key: string }
+import { WalletListSectionHeader } from './WalletListSectionHeader.js'
+import { WalletListSwipeRow } from './WalletListSwipeRow.js'
+
+type WalletListItem = {
+  id: string | null,
+  fullCurrencyCode?: string,
+  key: string,
```
Comment (amanalili) [inline src/components/themed/WalletList.js:28]: onLongPress?: function

File: src/components/themed/WalletList.js [hunk 2]
```diff
@@ -165,147 +191,158 @@ class WalletListComponent extends React.PureComponent<Props> {
...
+      const { createWalletType, createTokenType } = data.item
+      return <WalletListCreateRow createWalletType={createWalletType} createTokenType={createTokenType} onPress={onPress ?? selectWallet} />
+    }
+
     const walletId = data.item.id.replace(/:.*/, '')
     const guiWallet = wallets[walletId]
 
     if (guiWallet == null || !data.item.fullCurrencyCode) {
-      return <WalletListEmptyRow walletId={walletId} swipeRow={rowMap[data.item.key]} />
+      return <WalletListEmptyRow walletId={walletId} isModal={isModal} swipeRow={rowMap[data.item.key]} />
```
Comment (amanalili) [inline src/components/themed/WalletList.js:252]: pass onLongPress, gradiant

File: src/components/themed/WalletList.js [hunk 3]
```diff
@@ -165,147 +191,158 @@ class WalletListComponent extends React.PureComponent<Props> {
...
+        <WalletListSwipeRow
           currencyCode={currencyCode}
-          differencePercentage={differencePercentageString}
-          differencePercentageStyle={differencePercentageStringStyle}
-          exchangeRate={exchangeRateString}
-          exchangeRateFiatSymbol={exchangeRateFiatSymbol}
-          fiatBalance={fiatBalanceString}
-          fiatBalanceSymbol={fiatBalanceSymbol}
+          guiWallet={guiWallet}
+          isModal={isModal}
```
Comment (amanalili) [inline src/components/themed/WalletList.js:261]: pass onLongPress, gradiant

File: src/components/themed/WalletListCreateRow.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  }
+
+  render() {
+    const { createWalletType, createTokenType } = this.props
+
+    return (
+      <WalletListRow
+        currencyCode={createWalletType?.currencyCode ?? createTokenType?.currencyCode ?? ''}
+        icon={createWalletType?.symbolImage ?? createTokenType?.symbolImage ?? ''}
+        isModal
```
Comment (amanalili) [inline src/components/themed/WalletListCreateRow.js:122]: remove isModal

**Review by amanalili — COMMENTED** [review]
Create a new component WalletListCurrencyRow that renders WalletListRow with Icon, Rates and Balance

Add props showRates onPress and onLongPress and walletId and currencyCode

File: src/components/themed/WalletListEmptyRow.js [hunk 5]
```diff
@@ -25,42 +26,30 @@ class WalletListEmptyRowComponent extends React.PureComponent<Props & ThemeProps
...
           <TouchableOpacity style={styles.swipeButton} onPress={this.handleOpenWalletListMenuModal}>
             <EdgeText style={styles.swipeIcon}>{WALLET_LIST_OPTIONS_ICON}</EdgeText>
           </TouchableOpacity>
         </View>
-        <Gradient style={styles.container}>
-          <TouchableOpacity onLongPress={this.handleOpenWalletListMenuModal}>
-            <ActivityIndicator color={theme.primaryText} size="large" />
-          </TouchableOpacity>
-        </Gradient>
+        <WalletListRow currencyCode="" isModal={isModal} onLongPress={this.handleOpenWalletListMenuModal} walletId={walletId} walletName="" />
```
[the remark this answers is not in the document]
Comment (paullinator) [reply src/components/themed/WalletListEmptyRow.js]: @amanalili Sure that sounds reasonable.

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**